### PR TITLE
Add conformance tests

### LINKED ISSUE #31: Add conformance tests against latest releases of pack and lifecycle

### PR DESCRIPTION
**Pull request by jromero**
* Add kotlin-gradle app
* Rename java-gradle to kotlin-gradle to expand language coverage
* Add travis configuration

Resolves #31 

Signed-off-by: Javier Romero <jromero@pivotal.io>

### CODE REVIEW

File: apps/kotlin-gradle/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+	id("org.springframework.boot") version "2.1.8.RELEASE"
```
Comment (jkutner) [inline apps/kotlin-gradle/build.gradle.kts]: i wonder about version ranges here again.

File: apps/kotlin-gradle/src/main/resources/templates/index.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Buildpacks.io Java Sample</title>
```
Comment (jkutner) [inline apps/kotlin-gradle/src/main/resources/templates/index.html]: "Kotlin Gradle"?

File: apps/kotlin-gradle/build.gradle.kts (hunk 1 above)
Comment (jromero) [reply apps/kotlin-gradle/build.gradle.kts]: I researched it and couldn't find any way to do it. I experimented as well to no avail. I'm going with it's not possible at the moment but maybe someone else could find a solution.

Comment (jkutner) [reply apps/kotlin-gradle/build.gradle.kts]: It's fine if we can't. But we'll have to try and stay on top of updating it (I'm having flashbacks to the Struts example repos that used versions with the Equifax CVE for months after it went public).

File: buildpacks/kotlin-gradle/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Sample: Kotlin Gradle Buildpack
+
+Compatible apps:
+- Kotlin apps that use Gradle
+
+### Usage
+
+```bash
+pack build sample-app --builder <builder> --buildpack . --path ../../apps/kotlin-gradle
```
Comment (jkutner) [inline buildpacks/kotlin-gradle/README.md]: i'm not sure if we want to recommend this here (an arbitrary builder won't just work unless the buildpack is compatible with its stack). Maybe we should just point them to the main docs?

Comment (jromero) [reply buildpacks/kotlin-gradle/README.md]: As discussed, I think we might go with pointing to a sample-builder to guarantee compatibility and either make it a prerequisite or have it available in docker hub.

**Comment by jromero** [comment]
> I'm worried that we'll have trouble keeping the example app up-to-date, but I don't think that should block this.

If we can't auto depend on latest fixes maybe we could setup a vetting process on the apps that run as part of CI as well. Something like OWASP dependency check: https://jeremylong.github.io/DependencyCheck/dependency-check-gradle/index.html

**Pull request merged**